src/app/view/components/_common/dot-add-to-bundle/dot-add-to-bundle.component.ts

```diff
@@ -0,0 +1,101 @@
+import { BehaviorSubject, Observable } from 'rxjs/Rx';
+import { Component, Input, Output, EventEmitter, ViewEncapsulation, ViewChild } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { OnInit } from '@angular/core/src/metadata/lifecycle_hooks';
+import { DotMessageService } from '../../../../api/services/dot-messages-service';
+import { LoggerService } from 'dotcms-js/dotcms-js';
+import { AddToBundleService } from '../../../../api/services/add-to-bundle/add-to-bundle.service';
+import { DotBundle } from '../../../../shared/models/dot-bundle/dot-bundle';
+
+@Component({
+    encapsulation: ViewEncapsulation.None,
+    selector: 'dot-add-to-bundle',
+    templateUrl: 'dot-add-to-bundle.component.html'
+})
+export class DotAddToBundleComponent implements OnInit {
+    form: FormGroup;
+    bundle$: Observable<DotBundle[]>;
+    placeholder: string;
+
+    @Input() assetIdentifier: string;
+    @Output() cancel = new EventEmitter<boolean>();
+    @ViewChild('formEl') formEl: HTMLFormElement;
+
+    constructor(
+        private addToBundleService: AddToBundleService,
+        public fb: FormBuilder,
+        public dotMessageService: DotMessageService,
+        public loggerService: LoggerService
+    ) {}
+
+    ngOnInit() {
+        this.bundle$ = this.addToBundleService.getBundle();
+
+        this.dotMessageService.getMessages([
+            'contenttypes.content.add_to_bundle',
+            'contenttypes.content.add_to_bundle.select',
+            'contenttypes.content.add_to_bundle.type',
+            'contenttypes.content.add_to_bundle.errormsg',
+            'contenttypes.content.add_to_bundle.form.cancel',
+            'contenttypes.content.add_to_bundle.form.add'
+
+        ]).subscribe();
+
+        // tslint:disable-next-line:max-line-length
+        this.placeholder = this.bundle$ ? this.dotMessageService.get('contenttypes.content.add_to_bundle') : this.dotMessageService.get('contenttypes.content.add_to_type');
```

Comment:
Prettier format that so the line is not that big.

I think ```this.bundle$``` is ALWAYS going to be truthy because is an Observable... also I didn't see a test for this placeholder, you sure this works as expected?